### DISCUSSION TITLE
Filters for Document Library

### DISCUSSION THREAD

**Discussion by KaveSys** [microsoft-search/pnp-modern-search#3733] [Q&A] [answered]
Hello,
I'm trying to configure the PNP Filter by adding a checkbox to add document library as filterable, but also searching here/on the web, there isn't a way to do it.

From the home page, I add PNP SearchBox, PNP Search result and PNP Search Filter. The Sharepoint is: https://example.sharepoint.com/sites/abcdef

Below this site, there are multiple document library like:

https://example.sharepoint.com/sites/abcdef/test1/Forms/AllItems.aspx
https://example.sharepoint.com/sites/abcdef/test2/Forms/AllItems.aspx
https://example.sharepoint.com/sites/abcdef/test3/Forms/AllItems.aspx

There is a way to filter for document library, using the name/title of the document library? Like:

test1
test2
test3

Below my query text in PNP Search results:

{searchTerms} Path:https://example.sharepoint.com/sites/abcdef AND (contentclass:STS_ListItem OR contentclass:STS_List OR contentclass:STS_List_DocumentLibrary -Title:"Site Assets" AND -Title:"Form Templates" AND -Title:"Style Library"AND -Title:"Documents" AND -Title:"Site Pages") -filetype:aspx

**Comment by kasperbolarsen** [comment]
Are you talking about a query like this?
{searchTerms} (Path:https://example.sharepoint.com/sites/abcdef/test1* OR
Path:https://example.sharepoint.com/sites/abcdef/test2* OR  Path:
https://example.sharepoint.com/sites/abcdef/test3*)

Den ons. 8. maj 2024 kl. 14.50 skrev KaveSys ***@***.***>:

> Hello,
> I'm trying to configure the PNP Filter by adding a checkbox to add
> document library as filterable, but also searching here/on the web, there
> isn't a way to do it.
>
> From the home page, I add PNP SearchBox, PNP Search result and PNP Search
> Filter. The Sharepoint is: https://example.sharepoint.com/sites/abcdef
>
> Below this site, there are multiple document library like:
>
> https://example.sharepoint.com/sites/abcdef/test1/Forms/AllItems.aspx
> https://example.sharepoint.com/sites/abcdef/test2/Forms/AllItems.aspx
> https://example.sharepoint.com/sites/abcdef/test3/Forms/AllItems.aspx
>
> There is a way to filter for document library, using the name/title of the
> document library? Like:
>
> test1
> test2
> test3
>
> Below my query text in PNP Search results:
>
> {searchTerms} Path:https://example.sharepoint.com/sites/abcdef AND
> (contentclass:STS_ListItem OR contentclass:STS_List OR
> contentclass:STS_List_DocumentLibrary -Title:"Site Assets" AND -Title:"Form
> Templates" AND -Title:"Style Library"AND -Title:"Documents" AND
> -Title:"Site Pages") -filetype:aspx
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft-search/pnp-modern-search/discussions/3733>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE5DXIQDVISE6WHOFAMRCDDZBINTBAVCNFSM6AAAAABHM5EK7OVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZWGYZDQMZRGE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Med venlig hilsen/ Best regards

Kasper Bo Larsen

**Reply by KaveSys** [reply]
Oooh, thanks for the quick reply!

So I have to use the above Query text suggested and not based on the site on top? Then on the filter which filter field I have to use?

**Reply by KaveSys** [reply]
Let me explain better: this filter should be available in PNP Search Filter

**Reply by kasperbolarsen** [reply]
To me it sounds like either you are looking into using the library names as refiners or as Search suggestions ?

**Reply by KaveSys** [reply]
Not a search suggestion maybe, I just want to add the possibility for the users to filter on PNP Search Filter based on document library.
So If I try to search something and I want to search only in document library https://example.sharepoint.com/sites/abcdef/test1/Forms/AllItems.aspx for example, just select test1 on PNP Search Filter checkbox

**Comment by kasperbolarsen** [comment, marked as answer]
In that case you could map a RefinableString to OWS_Listtitle:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/3b38c261-3ddd-4391-97fc-4b82724a72c7)

wait for it to take effect
connect the refiner wp to the Results wp and likewise the other way round
add that managed prop to "selected properties" in the result web part, add it as a refiner in the refiner web part 

And you will have something like this ( you might want to trim away the "site pages" and the like) 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/17e5511a-ea67-4f93-8542-5f942b8c5b95)

**Reply by KaveSys** [reply]
Is working! Thanks a lot 👍

**Comment by Erik266-dot** [comment]
Hello, I (newbie / no code) might have a similar challenge: 

- For our Intranet on Sharepoint I have installed and configured PnP Search on a search results page 
- Either site pages (aspx) or files from a document library are being displayed. 
- This works so far. 
- One thing: instead of displaying the property 'createdby' I would like to show a custom property: the label 'owner', which I added as a category in the document library. This is a multiple choice list with employees to select the person responsible for the file from. Since not everyone has the rights to upload files to the document library I have uploaded some files for other colleagues. But it adds to the value of search results, if the true owner of a file is being displayed.
 
How do I add this property? And is it possible to add this as a filter in PnP filters?

Fresh regards,
Erik

**Reply by kasperbolarsen** [reply]
Hi Erik,

In general choice columns are a PITA to work with in search, and if your choice column did allow multiple values,  then it would be even worse. 
Had you created the "owner" field as a site column of type Person, then you would only had to map it's crawled property to a RefinableString,  and you were done. 
If you prefer the choice column approach,  then I would recommend finding a consultant with some Search experience,  as it is somewhat difficult.
